tesla_ver/data_uploading/data_uploading.py

```diff
@@ -0,0 +1,115 @@
+import dash
+import redis
+
+import pandas as pd
+import numpy as np
+import pyarrow as pa
+
+
+from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
+from base64 import b64decode
+from io import StringIO
+
+from tesla_ver.data_uploading.data_uploading_layout import LAYOUT
+from tesla_ver.redis_manager import redis_manager
+
+
+def generate_data_uploading(server):
+    app = dash.Dash(__name__, server=server, url_base_pathname="/datauploading.html/")
+    app.layout = LAYOUT
+
+    @app.callback(
+        [
+            Output("data-table-div", "style"),
+            Output("data-table-visualization", "data"),
+            Output("data-table-visualization", "columns"),
+        ],
+        [Input("upload", "contents"), Input("upload", "filename")],
+    )
+    def upload_data_parsing(contents, filename):
+        def upload_string_to_df(content):
```

Comment:
Is it necessary for this to be a local function? If not it should be defined elsewhere

Reply:
moved to `/data_uploading/utils.py` 5a1d5c3